tests/appsec/test_processor.py

```diff
@@ -68,18 +67,43 @@ def test_valid_json(tracer):
     _enable_appsec(tracer)
 
     with tracer.trace("test", span_type=SpanTypes.WEB) as span:
-        span.set_tag("http.url", "http://example.com/.git")
-        span.set_tag("http.status_code", "404")
+        set_http_meta(span, {}, raw_uri="http://example.com/.git", status_code="404")
 
     assert "triggers" in json.loads(span.get_tag("_dd.appsec.json"))
 
 
+def test_headers_collection(tracer):
+    _enable_appsec(tracer)
+
+    class Config(object):
+        def __init__(self):
+            self.is_header_tracing_configured = False
+
+    with tracer.trace("test", span_type=SpanTypes.WEB) as span:
+
+        set_http_meta(
+            span,
+            Config(),
+            raw_uri="http://example.com/.git",
+            status_code="404",
+            request_headers={
+                "hello": "world",
+                "accept": "something",
+                "x-Forwarded-for": "127.0.0.1",
+            },
+        )
+
+    assert span.get_tag("http.request.headers.hello") is None
+    assert span.get_tag("http.request.headers.accept") == "something"
+    assert span.get_tag("http.request.headers.x-forwarded-for") == "127.0.0.1"
```

Comment:
```suggestion
    assert span.get_tag("http.request.headers.x_forwarded_for") == "127.0.0.1"
```

Reply:
For this one, we need to keep the `-` if I understand https://github.com/DataDog/architecture/blob/master/rfcs/apm/integrations/trace-http-headers/rfc.md#tag-name well